src/compiler/checker.ts

```diff
@@ -9634,6 +10188,11 @@ namespace ts {
                 }
             }
 
+            // Handle 'define' calls
+            if (node.parent.kind === SyntaxKind.CallExpression && isDefineCall(<CallExpression>node.parent)) {
+                assignDefineOrRequireCallParameterTypes(<FunctionExpression>node);

```

Comment:
See my previous comment regarding pull vs. push. We should use our already existing contextual typing logic to provide contextual types for the `define` callback parameters.